java/cuvs-java/src/main/java/com/nvidia/cuvs/CuVSHostMatrix.java

```diff
@@ -20,4 +20,77 @@
  */
 public interface CuVSHostMatrix extends CuVSMatrix {
   int get(int row, int col);
+
+  default CuVSHostMatrix toHost() {
+    return new CuVSHostMatrixDelegate(this);
+  }
+
+  default CuVSDeviceMatrix toDevice(CuVSResources resources) {
+    var deviceMatrix = CuVSMatrix.deviceBuilder(resources, size(), columns(), dataType()).build();
+    toDevice(deviceMatrix, resources);
+    return deviceMatrix;
+  }
+
+  class CuVSHostMatrixDelegate implements CuVSHostMatrix {
+    private final CuVSHostMatrix hostMatrix;
```

Comment:
I've not sure I've complete grokked this pattern;  The `CuVSHostMatrixDelegate` already implements `CuVSHostMatrix`.  Why use composition alongside inheritance?

Specifically, would it be viable not to override `get()`, `size()`, `toArray()`, etc?  Instead, perhaps we could leave those unimplemented, and bank on the implementation in `CuVSHostMatrix`?  Maybe we could implement only `close()` as a NOOP.

Reply:
I think you are onto something here: since this is a inner class, we could use the reference to the inner class. But it would be effectively the same. In fact, I'd rather keep this obvious and explicit, make this a static class / not a inner class and pass the reference to the inner matrix explicitly